### ISSUE TITLE
Non Responsive Devices in HomeKit

### ISSUE THREAD

**Issue by themick4u** [hoobs-org/hoobsd#35]
After updating from 4.0.39 all bridged devices not responding in homekit. Accessories seem to work from within the hoobs cp without issue.

**Comment by mkellsy** [comment]
I tested and I am unable to reproduce.
Is this happening on all bridges? Do you have more than one bridge?
What plugins?

Try rebooting, this will clear the memory.
If that doesn't work, clear the bridge cache and re-pair.

**Comment by themick4u** [comment]
Yes all bridges, 4 plugins on separate bridges. Hubitat maker api, camera-ffmpeg, myQ and dummy
My HomeKit has a mix of both homebridge/hoobs devices as well as normal apple HomeKit devices. I will clear and re-pair.

**Comment by mkellsy** [comment]
Let me know. I have a similar setup and my HomeKit is works ng like a champ.

**Comment by themick4u** [comment]
I cleared cache and re-paired all 4 bridges and re-added to HomeKit.. working great and so far the memory is also looking good.

**Comment by mkellsy** [comment]
I'm going to close this for now. If it happens again we can either re-open or create a new issue.

**Comment by mkellsy** [comment]
This happened to me. I think the accessory cache is being deleted.

I have a fix for this coming in 4.0.42

**Comment by saiajin** [comment]
Yes continuing to have same issue on 40.41.....

**Comment by mkellsy** [comment]
I have 4.0.42 released. This one is a tricky one, and I am glad we found it in beta.

There was a conflict with the accessory_identifier field, that was used by HomeKit, HB and HOOBS.
The HOOBS identifier is a bit different, because it includes bridge information, so we can differentiate accessories when combined.
This accessory_identifier was causing the no response issue, because HomeKit was using the HOOBS id, but HB was caching the un-modified one.

To fix this, I did the same thing that plugins do, I created custom characteristics (Community Types) for our device_id and then I include the bridge_id and plugin_id too. So now this information is always there, and we can count on it.
Now, HB can cache the accessory_identifier and map it to the HomeKit associated accessory.

Because of this, after you upgrade, all of your accessories will disappear from HOOBS, and the accessories will remain unresponsive in HomeKit.
DONT WORRY! Everything is still there, the HB cache is corrupt now. It can be fixed.

To recover, you will need to clear the cache on all bridges.
Go to the bridges screen
Select the first bridge
Click Cache
Click Clear
Click OK
Bridge will restart
Repeat steps 2-6 for each bridge
Re-Pair bridges with HomeKit

After this, HomeKit will work even when HOOBS restarts.
Sorry for the inconvenience, but this is why we are in beta.
So thank You all for testing, it is helping to make HOOBS 4 a super solid product!
